### PR TITLE
Update mamba to 1.5.7 from 1.5.1, and conda to 24.3.0 from 23.7.4

### PR DESCRIPTION
**Pull request by SylvainCorlay**

### CODE REVIEW

**Comment by manics** [comment]
Looks like you'll need to bump conda too
```
INFO     repo2docker:app.py:869 error    libmamba Could not solve for environment specs
    The following packages are incompatible
    ├─ conda 23.7.4**  is requested and can be installed;
    └─ mamba 1.5.7**  is not installable because it requires
       └─ conda >=24,<25 , which conflicts with any installable versions previously reported.
`|`

File: tests/conda/py35-binder-dir/verify [hunk 1]
```diff
@@ -16,12 +16,12 @@ v = out.split()[1]
...
 out = sh(["micromamba", "--version"])
-assert out == "1.5.1", out
+assert out == "1.5.7", out
 
 out = sh(["mamba", "--version"])
 assert (
     out
-    == """mamba 1.5.1
+    == """mamba 1.5.7
 conda 23.7.4"""
```
Comment (consideRatio) [inline tests/conda/py35-binder-dir/verify:25]:
```suggestion
conda 24.3.0"""
```

File: repo2docker/buildpacks/conda/install-base-env.bash [hunk 2]
```diff
@@ -5,7 +5,7 @@ set -ex
 
 cd $(dirname $0)
 
-export MAMBA_VERSION=1.5.1
+export MAMBA_VERSION=1.5.7
 export CONDA_VERSION=23.7.4
```
Comment (consideRatio) [inline repo2docker/buildpacks/conda/install-base-env.bash:9]:
```suggestion
export CONDA_VERSION=24.3.0
```

**Comment by consideRatio** [comment]
1337 PR - thank you @SylvainCorlay!

**Pull request merged**